src/averageAges.js

```diff
@@ -37,7 +41,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const hoIsWoman = people.filter(person => person.sex === 'f');
+
+  const womenFiltr = !withChildren
+    ? hoIsWoman
+    : hoIsWoman.filter(
+      mathers => people.find(child => child.mother === mathers.name)
```

Comment:
There are no english word `mathers`, and here stored only one mother object.  Method `some` would be better for this purpose instead of  `find`, because here you need only boolean, not entire object.


```suggestion
      mother => people.find(child => child.mother === mathers.name)
```